functions.cpp

```diff
@@ -1566,6 +1643,235 @@ namespace Sass {
+        Selector_List* sel = p.parse_selector_group();
+        
+        parsedSelectors.push_back(sel);
+      }
+      
+      // Nothing to do
+      if( parsedSelectors.empty() ) {
+        return new (ctx.mem) Null(pstate);
+      }
+      
+      // Set the first element as the `result`, keep appending to as we go down the parsedSelector vector.
+      std::vector<Selector_List*>::iterator itr = parsedSelectors.begin();
+      Selector_List* result = *itr;
+      ++itr;
+      
+      for(;itr != parsedSelectors.end(); ++itr) {
+        Selector_List* child = *itr;
+        vector<Complex_Selector*> newElements;
+
+        // For every COMPLEX_SELECTOR in `child`
+        // For every COMPLEX_SELECTOR in `result`
+            // let parentSeqClone equal a copy of result->elements[i]
+            // let childSeq equal child->elements[j]
+              // Set childSeq as the new innermost tail of parentSeqClone
+            // Add parentSeqClone to the newElements
+        // Replace result->elements with newElements
+        for (size_t i = 0, resultLen = result->length(); i < resultLen; ++i) {
+          for (size_t j = 0, childLen = child->length(); j < childLen; ++j) {
+            Complex_Selector* parentSeqClone = (*result)[i]->cloneFully(ctx);
+            Complex_Selector*  childSeq = (*child)[j];
+            
+            parentSeqClone->innermost()->tail(childSeq); // Set seq as the new tail of parentSeqClone
+            newElements.push_back(parentSeqClone);
+          }
+        }
+      
+        result->elements(newElements);
+      }
+      
+     
+      Listize listize(ctx);
+      return result->perform(&listize);
+    }
+    
+    Signature selector_append_sig = "selector-append($selectors...)";
+    BUILT_IN(selector_append)
+    {
+      To_String to_string;
+      List* arglist = ARG("$selectors", List);
+      
+      // Not enough parameters
+      if( arglist->length() == 0 )
+        error("$selectors: At least one selector must be passed", pstate);
+      
+      // Parse args into vector of selectors
+      vector<Selector_List*> parsedSelectors;
+      for (size_t i = 0, L = arglist->length(); i < L; ++i) {
+        Expression* exp = dynamic_cast<Expression*>(arglist->value_at_index(i));
+        create_sel_parser(p_contextualize);
+        Selector_List* sel = p.parse_selector_group();
+        
+        parsedSelectors.push_back(sel);
+      }
+      
+      // Nothing to do
+      if( parsedSelectors.empty() ) {
+        return new (ctx.mem) Null(pstate);
+      }
+      
+      // Set the first element as the `result`, keep appending to as we go down the parsedSelector vector.
+      std::vector<Selector_List*>::iterator itr = parsedSelectors.begin();
+      Selector_List* result = *itr;
+      ++itr;
+      
+      for(;itr != parsedSelectors.end(); ++itr) {
+        Selector_List* child = *itr;
+        vector<Complex_Selector*> newElements;
+        
+        // For every COMPLEX_SELECTOR in `result`
```

Comment:
This superfluous comment. Unless you think it offers some value, I doesn't to me.